File: examples/pca/example_pca2.R
library("mmstat4")
bank2 <- ghload("bank2.rds")
par(mfrow=c(1,2))
# do PCA (correlation)
pc <- prcomp(bank2, center=T, scale=T)
summary(pc)
plot(pc$sdev^2, type="b", main="Scree plot")
#
library("psych")
ppc<-principal(bank2, rotate="none")
ppc
scree(bank2)

File: examples/pca/example_pca2.py
import pandas as pd
from sklearn.preprocessing import StandardScaler
from sklearn.decomposition import PCA
data = r['mmstat4::ghload']("bank2.rds")
df   = pd.DataFrame(data)
# Correlation
scaler = StandardScaler(with_mean=True, with_std=True)
scaled_data = scaler.fit_transform(df)
pca = PCA()
pca.fit(scaled_data)
print(pca.explained_variance_ratio_)
print(pca.explained_variance_)
# Covariance
pca.fit(data)
print(pca.explained_variance_ratio_)
print(pca.explained_variance_)
